### DISCUSSION TITLE
Laravel

### DISCUSSION THREAD

**Discussion by silicahd** [givanz/Vvveb#45] [Ideas]
Please migrate to something more common like laravel and livewire. This would make it so much easier for people to develop plugins and make changes.

**Comment by givanz** [comment]
Unfortunately it's not possible to use a framework such as Laravel to build Vvveb.

* Laravel is not shared hosting friendly and this would limit hosting options.

* Deploying Laravel apps is more complex than just unzip and upload to FTP making it harder to install for regular users.

* VvvebJs page builder uses plain html and all Vvveb themes are working static websites, this approach allows any element to be customized and there is no limit to what can be edited.

  With Blade (or Twig) it would not be possible to have the same level of flexibility and it would make everything more complex and themes harder to develop and customize.

  With Vvveb you can download any html template from one of the templates marketplaces upload the zip in the themes section and directly start customizing the theme from the page builder by changing content or adding dynamic sections, it's that easy.

* Laravel event system is complex and limited, Vvveb event system is very simple and this allows easy plugin development. 

  Plugins are key for the extensibility of the platform and a simple and powerful plugin architecture is imperative.

* Vvveb is MVCC (Model View Controller Component) the Component part is necessary in Vvveb to allow dynamic components to be inserted anywhere in the pages and is not easy to implement in Laravel and would probably need hacks in the core.

* Vvveb uses SQL mapping to php (with sqlP) for maximum flexibility, Laravel Eloquent (or others orms such as doctrine) hides away the power, simplicity and flexibility of the SQL language and makes things harder for complex queries and anything beyond usual crud.

I think that because of the hosting limitations and the need to have some basic knowledge of php/composer/artisan etc Laravel CMS are not as popular as the top market CMS solutions that don't use such php frameworks.

I know that there are many developers familiar with Laravel because it offers a standard/common way to organize code and it's well documented making it easy for developers to have a base from which to start building.

With Vvveb once you check the code and become familiar with how things are organized and work you will notice that it's easier than Laravel to develop, for themes you don't even need to know php/blade etc, plain html/css is enough.

The developer documentation is only a small draft and will be expanded soon after the code is pushed to github, plugins are very simple to implement you can take a glance on some of the existing ones https://github.com/topics/vvveb-plugin

**Comment by WebCrew** [comment]
The features are so complex and comprehensive, especially when everything goes as it should and the beta switches to stable at some point. 

On the other hand, it couldn't be easier for the end user to customize a website the way they want. There is really no comparable CMS that makes it easier for admins to style the homepage, but at the same time includes complex eCommerce functions. 

All that's missing at some point would be a forum plugin then with the CMS we could even get community functions out of the CMS in addition to eCommerce ohh my god hahahah.

**Comment by givanz** [comment]
@WebCrew Thanks for feedback.

I'm testing the last ecommerce features added and overall testing and making the last preparations for 1.0 launch.

A forum plugin is planned and it will be done after the SEO and Multi Vendor plugins are finished.

**Reply by WebCrew** [reply]
I'm happy to hear that, then it's worth just waiting :) 

SEO and Multi Vendor understandably have priority and are also important features for my use. 

**Thank you** for your work and **good luck**.

**Reply by givanz** [reply]
Thank you :)
